### PR TITLE
Dockers for GPU testing

### PR DESCRIPTION
**Pull request by Borda**
## Before submitting

- [x] Was this discussed/approved via a GitHub issue? (no need for typos and docs improvements)
- [ ] Did you create/update your **configuration file**?
- [ ] Did you **add your config to CI** - GitHub action and Azure pipeline?
- [x] Are all integration **tests passing**?

## What does this PR do? \[optional\]

creating own docker image with PyTorch and `sudo`

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]

File: actions/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG PYTORCH_VERSION=1.8.1
+ARG CUDA_VERSION=11.1
+
+# https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes
+#FROM nvcr.io/nvidia/pytorch:${PYTORCH_VERSION}-py3
```
Comment (justusschock) [inline actions/Dockerfile:19]: Is there a reason for not using the NVCR images?

Comment (Borda) [reply actions/Dockerfile:19]: it is too large :D

File: .github/workflows/dockers.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: Building dockers
+
+on: # Trigger the workflow on push or pull request, but only for the master branch
```
Comment (rohitgr7) [inline .github/workflows/dockers.yml:3]:
```suggestion
on: # Trigger the workflow on push or pull request, but only for the main branch
```

File: .github/workflows/dockers.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      - uses: docker/setup-buildx-action@v1
+      - name: Login to DockerHub
+        if: ${{ github.repository_owner == 'PyTorchLightning' }}
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build and Publish Docker Hub
+        # publish master/release
```
Comment (rohitgr7) [inline .github/workflows/dockers.yml:34]:
```suggestion
        # publish main/release
```

**Review by rohitgr7 — APPROVED** [review]
nit:

**Comment by Borda** [comment]
> nit:

sorry, it was an automatic merge... :/

**Comment by rohitgr7** [comment]
> > nit:
> 
> sorry, it was an automatic merge... :/

I think you should configure the `no-merge until all conversations are resolved` plugin here.

**Comment by Borda** [comment]
> I think you should configure the `no-merge until all conversations are resolved` plugin here.

it was already set, so probably something fishy about the rule...

**Pull request merged**